src/scripts/settings/Settings.ts

```diff
@@ -0,0 +1,59 @@
+/// <reference path="Setting.ts" />
+/// <reference path="BooleanSetting.ts"/>
+/// <reference path="MultipleChoiceSetting.ts"/>
+
+class Settings {
+    static list: Setting[] = [];
+
+
+    static add(setting: Setting) {
+        if (this.list.indexOf(setting) === -1) {
+            this.list.push(setting);
+        }
+    }
```

Comment:
`this.list.indexOf(setting)` will only not be `-1` if the object `setting` has already been put in the list. A different object with identical properties will not match.

I think what we really want to be checking here is whether there is already a setting with the same name, as that would cause a conflict with setting/loading.
eg, `if (!this.getSetting(setting.name))`

```
x = {'a':1}
y = {'a':1}
list = []
list.push(x)
list.indexOf(x)
> 0
list.indexOf(y)
> -1
```

Reply:
Interesting, fixed